### PR TITLE
[driver/index] Fix an issue where the -index-file invocation was not working when there are 5 or more input files

### PR DESCRIPTION
**Pull request by akyrtzi**
When the supplementary-outputs file is written for an -index-file invocation it writes out the first input file
given in arguments instead of the input file that is provided by -index-file-path option. This then causes the frontend
index invocation to fail with an error because there is a mismatch in the primary input file and the input file that is written
in the supplementary-outputs file.

Addresses rdar://39398871

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci smoke test

File: lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -2217,6 +2217,16 @@ Job *Driver::buildJobsForAction(Compilation &C, const JobAction *JA,
     PrimaryInput = Out.getPrimaryOutputFilenames()[i];
   }
 
+  // With -index-file option, the primary input is the one passed with
+  // -index-file-path.
```
Comment (davidungar) [inline lib/Driver/Driver.cpp]: Would it be possible to add a pointer here to a larger comment or a document that describes how -index-file-path works? How it is used, or how such an invocation differs from normal? I don't really understand this stuff and don't know where to look and the next person who has to work on the driver may have the same problem. Thanks.

**Review by davidungar — APPROVED** [review]
This looks fine, but it would be great if you could add something to the new comment in Driver.cpp.

**Comment by jrose-apple** [comment]
@nathawes' change should make it so we don't use a supplementary output file map at all, right? Or does that not apply to -index-file mode?

The -index-ignore-system-modules change seems unrelated.

**Comment by benlangmuir** [comment]
@jrose-apple -index-file is a `swiftc` invocation, not a sourcekit thing, so it won't have a custom threshold.

> The -index-ignore-system-modules change seems unrelated.

The -index-ignore-system-modules change was added here so that we could have an end-to-end test for -index-file from the driver.  Without this the test would trigger indexing the standard library, which is very expensive for a test.  Do you have any concerns about the flag?

**Comment by jrose-apple** [comment]
Well, I think the "DoesNotAffectIncrementalBuild" is wrong, but other than that no, it seems fine.

**Comment by benlangmuir** [comment]
Ah, good point. I'll fix that.

**Comment by benlangmuir** [comment]
@swift-ci please smoke test

File: lib/Driver/Driver.cpp (hunk 1 above)
Comment (benlangmuir) [reply lib/Driver/Driver.cpp]: +1 for more documentation of how this mode is supposed to work.  I admit I don't fully understand it either.  I will talk to @akyrtzi  about addressing this as a follow-up commit.

**Pull request merged**